kornia/geometry/plane.py

```diff
@@ -3,104 +3,129 @@
+        KORNIA_CHECK_TYPE(n, Vector3)
+        KORNIA_CHECK_TYPE(d, Scalar)
+        # TODO: fix checkers
+        # KORNIA_CHECK_SHAPE(n, ["B", "*"])
+        # KORNIA_CHECK_SHAPE(d, ["B"])
+        self._n: Vector3 = n
+        self._d: Scalar = d
 
     def __str__(self) -> str:
         return f"Normal: {self.normal}\nOffset: {self.offset}"
 
     def __repr__(self) -> str:
         return str(self)
 
-    def __getitem__(self, idx) -> Tensor:
-        return self.normal if idx == 0 else self.offset
-
-    def __iter__(self):
-        yield from (self.normal, self.offset)
-
     @property
-    def normal(self) -> Tensor:
+    def normal(self) -> Vector3:
         return self._n
 
     @property
-    def offset(self) -> Tensor:
+    def offset(self) -> Scalar:
         return self._d
 
+    def abs_distance(self, p: Vector3) -> Scalar:
+        return Scalar(self.signed_distance(p).abs())
+
+    # https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h#L145
+    # TODO: tests
+    def signed_distance(self, p: Vector3) -> Scalar:
+        KORNIA_CHECK(isinstance(p, (Vector3, Tensor)))
+        return self.normal.dot(p) + self.offset
+
+    # https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Hyperplane.h#L154
+    # TODO: tests
+    def projection(self, p: Vector3) -> Vector3:
+        dist = self.signed_distance(p)
+        if len(dist.shape) != len(self.normal):
+            # non batched plane project a batch of points
+            dist = dist[..., None]  # Nx1
+        # TODO: TypeError: bad operand type for unary -: 'Scalar'
+        return p - dist.data * self.normal
+        # TODO: make that Vector can subtract Scalar
+        # return p - self.signed_distance(p) * self.normal
+
     @classmethod
-    def from_vector(self, n: Tensor, e: Tensor) -> "Hyperplane":
-        normal = n
-        offset = -batched_dot_product(normal, e)
-        return Hyperplane(normal, offset)
+    def from_vector(self, n: Vector3, e: Vector3) -> "Hyperplane":
+        normal: Vector3 = n
+        # TODO: implement as below
+        # offset: Scalar = -normal.dot(e)
+        offset: Scalar = -batched_dot_product(normal, e)
+        return Hyperplane(normal, Scalar(offset))
 
     @classmethod
     def through(cls, p0: Tensor, p1: Tensor, p2: Optional[Tensor] = None) -> "Hyperplane":
         # 2d case
         if p2 is None:
-            KORNIA_CHECK_SHAPE(p0, ["B", "2"])
-            KORNIA_CHECK_SHAPE(p1, ["B", "2"])
+            # TODO: improve tests
+            KORNIA_CHECK_SHAPE(p0, ["*", "2"])
+            KORNIA_CHECK(p0.shape == p1.shape)
             # TODO: implement `.unitOrthonormal`
-            normal = normalize((p1 - p0), p=2, dim=-1)
-            offset = -batched_dot_product(p0, normal)
-            return Hyperplane(normal, offset)
+            normal: Vector3 = normalized(p1 - p0)
+            offset: Tensor = -batched_dot_product(p0, normal)
+            return Hyperplane(normal, Scalar(offset))
```

Comment:
```suggestion
            normal2d = normalized(p1 - p0)
            offset2d = -batched_dot_product(p0, normal2d)
            return Hyperplane(Vector3(normal2d), Scalar(offset2d))
```

Reply:
i think next will be to implement `Vector2d`, possibly we need an intermediate class `VectorType` as i planned before based on eigen